waku/factory/waku.nim

```diff
@@ -101,16 +101,15 @@ proc validateShards(conf: WakuNodeConf): Result[void, string] =
 
 ## Initialisation
 
-proc init*(T: type Waku, conf: WakuNodeConf): Result[Waku, string] =
-  var confCopy = conf
+proc init*(T: type Waku, confCopy: var WakuNodeConf): Result[Waku, string] =
```

Comment:
I actually think these are the only needed changes.
Did you try to see what happens with just that?

Reply:
Yes, so the need of each change is the following:

1. In this file, apart from this line, changed the name to refer that we are using a copy of the original configuration and that we're not modifying the user's input

2. In `apps/wakunode2/wakunode2.nim`, we create a copy of the original configuration and pass it to all the procs (so we don't modify the original configuration passed by the user)

3. In `tests/wakunode2/test_app.nim`, had to change from `let` to `var` for the configurations passed to `init()` now that we modify the object in the proc

Reply:
Do we use/re-use the original user configuration afterward? Just curious...

Reply:
> Do we use/re-use the original user configuration afterward? Just curious...

After that point we actually don't use the original configuration, but did it just to not override the original user configuration which generally isn't a good pattern. In principle we can avoid doing the copy and change the original but idk, it doesn't feel right haha